app/validation.go

```diff
@@ -188,30 +308,22 @@ func checkForDuplicatesBetweenManifestAndLight(light map[string]string, roleMani
 
 	check := make(map[string]struct{})
 
-	// The global properties, ...
-	for _, propertyDef := range roleManifest.Configuration.Templates {
-		property := propertyDef.Key.(string)
-		template := propertyDef.Value.(string)
-
-		allErrs = append(allErrs, checkForDuplicateProperty("configuration.templates", property, template, light, true)...)
-		check[property] = struct{}{}
-	}
-
-	// ... then the per-instance-group properties
 	for _, instanceGroup := range roleManifest.InstanceGroups {
 		prefix := fmt.Sprintf("instance-groups[%s].configuration.templates", instanceGroup.Name)
 
-		for _, propertyDef := range instanceGroup.Configuration.Templates {
-			property := propertyDef.Key.(string)
-			template := propertyDef.Value
-
-			// Skip over duplicates of the global
-			// properties in the per-instance-group data, we already
-			// checked them, see above.
-			if _, ok := check[property]; ok {
-				continue
+		for property, template := range instanceGroup.Configuration.Templates {
+			if template.IsGlobal {
+				if _, ok := check[property]; ok {
+					// Skip over duplicates of the global properties in the
+					// per-instance-group data, we already checked them in a
+					// different role
+					continue
+				}
+				check[property] = struct{}{}
```

Comment:
Every time I see `struct{}{}` I find it to be the ugliest thing, but at the same time, I love what it does.